core/chains/evm/txmgr/config.go

```diff
@@ -44,35 +44,36 @@ type ListenerConfig interface {
 }
 
 type (
-	EvmTxmConfig         txmgrtypes.TransactionManagerConfig
-	EvmBroadcasterConfig txmgrtypes.BroadcasterConfig
-	EvmConfirmerConfig   txmgrtypes.ConfirmerConfig
+	EvmTxmConfig         txmgrtypes.TransactionManagerChainConfig
+	EvmTxmFeeConfig      txmgrtypes.TransactionManagerFeeConfig
+	EvmBroadcasterConfig txmgrtypes.BroadcasterChainConfig
+	EvmConfirmerConfig   txmgrtypes.ConfirmerChainConfig
 	EvmResenderConfig    txmgrtypes.ResenderChainConfig
-	EvmReaperConfig      txmgrtypes.ReaperConfig
+	EvmReaperConfig      txmgrtypes.ReaperChainConfig
 )
 
 var _ EvmTxmConfig = (*evmTxmConfig)(nil)
 
 type evmTxmConfig struct {
-	Config
+	ChainConfig
 }
 
-func NewEvmTxmConfig(c Config) *evmTxmConfig {
+func NewEvmTxmConfig(c ChainConfig) *evmTxmConfig {
 	return &evmTxmConfig{c}
 }
 
 func (c evmTxmConfig) IsL2() bool { return c.ChainType().IsL2() }
 
-func (c evmTxmConfig) MaxFeePrice() string { return c.EvmMaxGasPriceWei().String() }
-
-func (c evmTxmConfig) FeePriceDefault() string { return c.EvmGasPriceDefault().String() }
-
-func (c evmTxmConfig) FeeBumpTxDepth() uint32 { return c.EvmGasBumpTxDepth() }
+var _ EvmTxmFeeConfig = (*evmTxmFeeConfig)(nil)
 
-func (c evmTxmConfig) FeeLimitDefault() uint32 { return c.EvmGasLimitDefault() }
+type evmTxmFeeConfig struct {
+	FeeConfig
+}
 
-func (c evmTxmConfig) FeeBumpThreshold() uint64 { return c.EvmGasBumpThreshold() }
+func NewEvmTxmFeeConfig(c FeeConfig) *evmTxmFeeConfig {
+	return &evmTxmFeeConfig{c}
+}
 
-func (c evmTxmConfig) FinalityDepth() uint32 { return c.EvmFinalityDepth() }
+func (c evmTxmFeeConfig) MaxFeePrice() string { return c.PriceMax().String() }
 
-func (c evmTxmConfig) FeeBumpPercent() uint16 { return c.EvmGasBumpPercent() }
+func (c evmTxmFeeConfig) FeePriceDefault() string { return c.PriceDefault().String() }
```

Comment:
I'm surprised these are strings. Do we just re-parse them again? If so can we clean that up or open a ticket?

Reply:
Atm they're mainly used for logging. From what I understand this was done to generify the API and make it not dependent on the assets.Wei type (note: this wasn't introduced by me, it just appears that way because of the diff).

@aalu1418 or @prashantkumar1982 may have more context on this and whether it's intentional or an error.

Reply:
Ah that make sense, but does it also suggest an asymmetry where the system doing the logging of the values somehow doesn't actually use them? Shouldn't they be logged alongside the usage? :shrug: